react/components/MinicartIconButton.tsx

```diff
@@ -14,20 +14,42 @@ interface Props {
   itemCountMode: MinicartTotalItemsType
 }
 
-const totalItemsSum = (arr: OrderFormItem[]) =>
-  arr.reduce((sum: number, product: OrderFormItem) => sum + product.quantity, 0)
+const countModeHandle = (
+  countMode: MinicartTotalItemsType,
+  arr: OrderFormItem[]
+) => {
+  const options = {
+    distinctAvailable: arr.reduce(
+      (itemQuantity: number, item: OrderFormItem) => {
+        if (item.availability === 'available') {
+          return ++itemQuantity
+        }
+        return itemQuantity
+      },
+      0
+    ),
+    totalAvailable: arr.reduce((itemQuantity: number, item: OrderFormItem) => {
+      if (item.availability === 'available') {
+        return itemQuantity + item.quantity
+      }
+      return itemQuantity
+    }, 0),
+    total: arr.reduce((itemQuantity: number, item: OrderFormItem) => {
+      return itemQuantity + item.quantity
+    }, 0),
+    distinct: arr.length,
+  }
+
+  return options[countMode] ?? options.distinct
+}
```

Comment:
See the review comment here: https://github.com/vtex-apps/product-list/pull/82/files#r501052532